web/source/osk/activeLayout.ts

```diff
@@ -12,18 +14,25 @@ namespace com.keyman.osk {
     id?: string;
     width?: string;
     pad?: string;
+    layer: string;
+    displayLayer: string;
+    nextlayer: "string";
     
     proportionalX: number;
     proportionalWidth: number;
 
-    static polyfill(key: LayoutKey) {
+    static polyfill(key: LayoutKey, displayLayer: string) {
```

Comment:
I don't understand what the `displayLayer` is. Could you please clarify?

Reply:
Sure.  By default, each key on our keyboards derives its base modifiers from its layer, which is especially important for keyboards without touch-optimized versions.  For example, loosely speaking, any key on the 'shift' layer is treated as if a physical `SHIFT` key were already held down when pressed.

The issue is that we have some keyboards that not only make use of this property but that also manually set the 'layer' property for select keys - making those keys derive modifiers from their custom property instead of the layer they're actually on.  For an example, see the `sil_cameroon_azerty` keyboard and its touch-layout's shift layer.  Most of the keys on the top two rows are like this.

While that's good for keystroke processing, it's not helpful when trying to find potential fat-finger candidates.  We want to find other keys from the same _display layer_, not with the same physical keyboard modifiers - but `LayoutKey` was only storing the 'layer' property corresponding to the latter.  Hence, `displayLayer` to fill that gap.